src/Statistics/Histogram.cpp

```diff
@@ -54,4 +56,19 @@ static Histogram BuildHistogramWithNumberOfBins(const std::optional<DataSet>& da
   return best_histogram;
 }
 
+[[nodiscard]] std::vector<int> GetBinWidth(size_t number_of_bins, int histogram_width) {
+  const int narrower_width = histogram_width / number_of_bins;
+  const int wider_width = narrower_width + 1;
+
+  const int number_of_wider_bins = histogram_width % number_of_bins;
+  const int number_of_narrower_bins = number_of_bins - number_of_wider_bins;
+
+  std::vector<int> result(number_of_narrower_bins, narrower_width);
+  const std::vector<int> wider_widths(number_of_wider_bins, wider_width);
+  result.insert(std::end(result), std::begin(wider_widths), std::end(wider_widths));
+
+  std::shuffle(std::begin(result), std::end(result), std::default_random_engine{});
```

Comment:
So you use this function to get the width in pixels for each bin. Each such width should be "equalisch", but for sure some may be off by 1 due to being integers.

And now you do the shuffling such those bins that are 1 pixel different are distributed across the histogram, right?

I think the shuffling might deserve a brief comment.

Reply:
Yep, you're right.